gallery/src/styles/NxFormLayout/NxFormLayoutExample.tsx

```diff
@@ -4,61 +4,93 @@
+          <NxStatefulTextInput className="nx-text-input--long"/>
         </label>
       </div>
       <fieldset className="nx-fieldset">
-        <legend className="nx-label">Checkboxes</legend>
-        <NxCheckbox isChecked={false}>Checkbox 1</NxCheckbox>
-        <NxCheckbox isChecked={true}>Checkbox 2</NxCheckbox>
-        <NxCheckbox isChecked={false}>Checkbox 3</NxCheckbox>
+        <legend className="nx-legend">
+          <span className="nx-legend__text">
+            Checkboxes
+          </span>
+        </legend>
+        <NxCheckbox onChange={toggleRed} isChecked={isRed}>Red</NxCheckbox>
+        <NxCheckbox onChange={toggleBlue} isChecked={isBlue}>Blue</NxCheckbox>
+        <NxCheckbox onChange={toggleGreen} isChecked={isGreen}>Green</NxCheckbox>
       </fieldset>
       <fieldset className="nx-fieldset">
-        <legend className="nx-label">Radio buttons</legend>
-        <NxRadio name="demo1" value="demo1" isChecked={true}>Radio Button 1</NxRadio>
-        <NxRadio name="demo2" value="demo2" isChecked={false}>Radio Button 2</NxRadio>
-        <NxRadio name="demo3" value="demo3" isChecked={true}>Radio Button 3</NxRadio>
+        <legend className="nx-legend"><span className="nx-legend__text">Radio buttons</span></legend>
+        <NxRadio name="color"
+                 value="red"
+                 onChange={setColor}
+                 isChecked={color === 'red'}
+                 radioId="color-red">
+          Red
+        </NxRadio>
+        <NxRadio name="color"
+                 value="purple"
+                 onChange={setColor}
+                 isChecked={color === 'purple'}
+                 radioId="color-purple">
+          Purple
+        </NxRadio>
+        <NxRadio name="color" value="green" onChange={setColor} isChecked={color === 'green'} radioId="color-green">
+          Green
+        </NxRadio>
+        <NxRadio name="color" value="blue" onChange={setColor} isChecked={color === 'blue'} radioId="color-blue">
+          Blue
+        </NxRadio>
       </fieldset>
       <div className="nx-form-group">
         <label className="nx-label">
           <span className="nx-label__text">Textarea</span>
-          <textarea className="nx-text-input"></textarea>
-        </label>
-      </div>
-      <div className="nx-form-group">
-        <label className="nx-label">
-          <span className="nx-label__text">Long Textarea</span>
-          <textarea className="nx-text-input nx-text-input--long"></textarea>
+          <NxStatefulTextInput type="textarea" placeholder="placeholder"/>
         </label>
       </div>
-      <div className="nx-btn-bar nx-btn-bar--forms">
-        <button className="nx-btn nx-btn--primary">Submit</button>
-        <button className="nx-btn">Cancel</button>
-      </div>
+      <footer className="nx-form-footer">
+        <div className="nx-btn-bar">
+          <NxButton type="button">Cancel</NxButton>
+          <NxButton variant="primary" type="button">Submit</NxButton>
```

Comment:
Like the other one, change it to have an `onSubmit` handler.

Reply:
> Like the other one, change it to have an `onSubmit` handler.

https://github.com/sonatype/sonatype-react-shared-components/pull/122/commits/09a0b0871980983fb119a6202677911e21261a13